File: 242-ValidAnagram/242-ValidAnagram.java
// Last updated: 14/07/2026, 14:31:24
class Solution {
    public boolean isAnagram(String s, String t) 
    {
        char arr[]=s.toCharArray();
        char arr1[]=t.toCharArray();
        Arrays.sort(arr);
        Arrays.sort(arr1);
        if(arr.length!=arr1.length)
        {
            return false;
        }
        // for(int i=0;i<arr.length;i++)
        // {
        //     System.out.println(arr[i]);
        // }
        // for(int i=0;i<arr.length;i++)
        // {
        //     System.out.println(arr1[i]);
        // }
        
        int i=0;
        while(i<arr.length)
        {
            
            if(arr[i]==arr1[i])
            {
                i++;
            }
            else
            {
                return false;
            }
        }
    return arr.length==i;  
    }
}

File: 242-ValidAnagram/242-ValidAnagram.py
# Last updated: 14/07/2026, 14:31:21
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        arr = sorted(s)
        arr1 = sorted(t)

        if len(arr) != len(arr1):
            return False

        i = 0
        while i < len(arr):
            if arr[i] == arr1[i]:
                i += 1
            else:
                return False

        return len(arr) == i
        